File: 1012 - area/area.lua
function areadotrianguloretangulo( baseA,  alturaC )
    return ((baseA * alturaC) / 2.0 )
end

function areadocirculo( raioC )
    return ( 3.14159 * raioC * raioC)
end

function areadotrapezio( baseA , baseB , alturaC )
    return (((baseA + baseB) * alturaC ) / 2.0 )
end

function areadoquadrado( ladoB )
    return (ladoB * ladoB)
end

function areadoretangulo( ladoA, ladoB )
    return (ladoA * ladoB)
end

a, b, c = io.read("*n", "*n", "*n")
atr = string.format('%.3f', areadotrianguloretangulo(a, c))
ac = string.format('%.3f', areadocirculo(c))
at = string.format('%.3f', areadotrapezio(a, b, c))
aq = string.format('%.3f', areadoquadrado(b))
ar = string.format('%.3f', areadoretangulo (a, b))
print("TRIANGULO: "..atr)
print("CIRCULO: "..ac)
print("TRAPEZIO: "..at)
print("QUADRADO: "..aq)
print("RETANGULO: "..ar)

File: 1012 - area/area.pas
program area( Input, Output);

uses 
    Sysutils;

function areadotrianguloretangulo( baseA: Double; alturaC: Double) : Double;
begin
    areadotrianguloretangulo:= ((baseA * alturaC) / 2.0 );
end;

function areadocirculo( raioC : Double ) : Double;
begin
    areadocirculo := ( 3.14159 * raioC * raioC);
end;

function areadotrapezio( baseA : Double ; baseB : Double ; alturaC : Double ) : Double;
begin
    areadotrapezio := (((baseA + baseB) * alturaC ) / 2.0 );
end;

function areadoquadrado( ladoB : Double ) : Double;
begin
    areadoquadrado := (ladoB * ladoB);
end;

function areadoretangulo( ladoA : Double; ladoB : Double ) : Double;
begin
    areadoretangulo := (ladoA * ladoB);
end;

var 
    a, b, c : Double;
    atr, ac, at, aq, ar : Double;

begin
    readLn (a, b, c);
    atr := areadotrianguloretangulo( a, c );
    ac := areadocirculo( c );
    at := areadotrapezio( a, b, c);
    aq := areadoquadrado( b );
    ar := areadoretangulo(a, b);

    writeln('TRIANGULO: ', Format('%.3f',[atr]));
    writeln('CIRCULO: ', Format('%.3f',[ac]));
    writeln('TRAPEZIO: ', Format('%.3f',[at]));
    writeln('QUADRADO: ', Format('%.3f',[aq]));
    writeln('RETANGULO: ', Format('%.3f',[ar]));

end.
